test/byContract/IexecPoco/03_reveal.test.ts

```diff
@@ -0,0 +1,291 @@
+                { signer: worker3, resultDigest: resultDigest },
+                { signer: worker4, resultDigest: resultDigest },
+            ];
+            const loosingWorker = worker2;
+            // winning workers are worker1, worker3 & worker4
+            for (let i = 0; i < workers.length; i++) {
+                const worker = workers[i];
+                const workerAddress = worker.signer.address;
+                const { resultHash, resultSeal } = buildResultHashAndResultSeal(
+                    taskId,
+                    worker.resultDigest,
+                    worker.signer,
+                );
+                await iexecWrapper.depositInIexecAccount(worker.signer, workerTaskStake);
+                await iexecPoco
+                    .connect(worker.signer)
+                    .contribute(
+                        taskId,
+                        resultHash,
+                        resultSeal,
+                        emptyEnclaveAddress,
+                        emptyEnclaveSignature,
+                        await buildAndSignContributionAuthorizationMessage(
+                            workerAddress,
+                            taskId,
+                            emptyEnclaveAddress,
+                            scheduler,
+                        ),
+                    )
+                    .then((tx) => tx.wait());
+            }
+            const task = await iexecPoco.viewTask(taskId);
+            expect(task.status).equal(TaskStatusEnum.REVEALING);
+            const contribution = await iexecPoco.viewContribution(taskId, loosingWorker.address);
+            expect(contribution.status).equal(ContributionStatusEnum.CONTRIBUTED);
+            expect(contribution.resultHash).not.equal(task.consensusValue);
+            // revealing task, before deadline, contribution status is contributed
+            // but contribution outside consensus
+            await expect(
+                iexecPoco.connect(loosingWorker).reveal(taskId, resultDigest),
+            ).to.be.revertedWithoutReason(); // require#4
+        });
+
+        it('Should not reveal when no able to prove result value', async () => {
+            await iexecPocoAsWorker
+                .contribute(
+                    taskId,
+                    resultHash,
+                    resultSeal,
+                    emptyEnclaveAddress,
+                    emptyEnclaveSignature,
+                    schedulerSignature,
+                )
+                .then((tx) => tx.wait());
+            const task = await iexecPoco.viewTask(taskId);
+            expect(task.status).equal(TaskStatusEnum.REVEALING);
+            const contribution = await iexecPoco.viewContribution(taskId, worker.address);
+            expect(contribution.status).equal(ContributionStatusEnum.CONTRIBUTED);
+            expect(contribution.resultHash).equal(task.consensusValue);
+            expect(contribution.resultHash).not.equal(buildResultHash(taskId, badResultDigest));
```

Comment:
Why this check?

Reply:
Since errors are not explicit:
https://github.com/iExecBlockchainComputing/PoCo/blob/934cff4b7706b3baa57a1b4c4b5f165af6cf78e6/contracts/modules/delegates/IexecPoco2Delegate.sol#L15
we emphasize what will go well and what will not:
https://github.com/iExecBlockchainComputing/PoCo/blob/934cff4b7706b3baa57a1b4c4b5f165af6cf78e6/contracts/modules/delegates/IexecPoco2Delegate.sol#L255

If you want we can add another temp var to be sure that `badResultDigest` is indeed used for the reveal here (?)

Reply:
I see. It's ok ne need to change it.